### DISCUSSION TITLE
Composer semantic commits

### DISCUSSION THREAD

**Discussion by bytestream** [renovatebot/renovate#8736] [Request Help]
**What Renovate type, platform and version are you using?**

Self-Hosted GitLab

**Describe the bug**

All commits generated by Renovate for the composer package manager use `chore(deps)`. For npm it defaults to `fix(deps)` for production and `chore(deps)` for development dependencies.

**Relevant debug logs**

```
DEBUG: Creating PR (repository=supportpal/integrations/whmcs-integration, branch=renovate/supportpal-api-client-php-1.x)
       "prTitle": "chore(deps): update dependency supportpal/api-client-php to v1.3.2"
```

**Have you created a minimal reproduction repository?**

I can create one if necessary but this may be a feature request rather than a bug.

**Additional context**

**Comment by viceice** [comment]
It is possible to change that. Please provide some more details about your config.

**Reply by bytestream** [reply]
I'm just using `semanticCommits: true,`

This works fine for the npm package manager, and exhibits different behaviour for the composer package manager.

**Reply by rarkins** [reply]
Provide your full config.

**Reply by bytestream** [reply]
```
module.exports = {
    platform: 'gitlab',
    endpoint: 'https://gitlab.x.com/api/v4/',
    token: process.env.RENOVATE_TOKEN,

    repositories: [process.env.RENOVATE_REPO],

    logLevel: 'debug',
    
    labels: ["dependencies"],

    requireConfig: true,
    onboarding: true,
    onboardingConfig: {
        extends: ['config:base'],
        prConcurrentLimit: 0,
        prHourlyLimit: 0,
    },

    enabledManagers: [
        'npm',
        'composer'
    ],

    composerIgnorePlatformReqs: false,
    
    dependencyDashboard: true,
    major: {
        "dependencyDashboardApproval": true
    },
    
    semanticCommits: true,

    hostRules: [
        {
            domainName: "github.com",
            token: process.env.GITHUB_COM_TOKEN
        },
        {
            hostType: "packagist",
            domainName: "gitlab.x.com",
            token: process.env.GITLAB_TOKEN
        },
        {
            hostType: "npm",
            domainName: "gitlab.x.com",
            token: process.env.GITLAB_TOKEN
        }
    ],
    
    packageRules: [
        {
            managers: ["npm"],
            labels: ["npm","dependencies"]
        },
        {
            managers: ["composer"],
            labels: ["composer","dependencies"]
        },
        {
            updateTypes: ["minor", "patch", "pin", "digest"],
            automerge: true
        }
    ],
};

```

**Reply by rarkins** [reply]
And how about the repository's `renovate.json`?

**Reply by bytestream** [reply]
Unchanged from the `onboardingConfig`
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "prConcurrentLimit": 0,
  "prHourlyLimit": 0
}

```

**Reply by rarkins** [reply]
You are using the config preset [config:base](https://docs.renovatebot.com/presets-config/#configbase), which includes [:semanticPrefixFixDepsChoreOthers](https://docs.renovatebot.com/presets-default/#semanticprefixfixdepschoreothers)

This is where the `fix` semantic commit type you see in npm comes from. In Composer the depType is `require` instead of `dependencies`.  You could write your own package rule for `require` or contribute a PR to this repo to include `require` in the existing rule.

**Reply by bytestream** [reply]
Thank you. I'll send a PR.
